src/m3ninx/index/segment/fst/segment.go

```diff
@@ -195,6 +206,24 @@ func (r *fsSegment) SegmentData(ctx context.Context) (SegmentData, error) {
 	return r.data, nil
 }
 
+func (r *fsSegment) Freeze() {
+	r.Lock()
+	defer r.Unlock()
+	if r.closed {
+		return
+	}
+	r.data.State = FrozenIndexSegmentState
+}
+
+func (r *fsSegment) State() IndexSegmentState {
```

Comment:
Maybe this should return an error if closed? Seems like a failed pre-condition. (i.e. should not be querying if closed)